### PR TITLE
Improve runtest defaults, add changelog, update --version flag, add --force flag, refactor Rlibs

### LINKED ISSUE #96: Use version file instead of git commit tag
assumes the CLI script is in the git repo, but that's not necessarily true after installation

### LINKED ISSUE #97: Consider allowing workdir to already existing during `init`
`carlisle --runmode=init --workdir=.`

```
Pipeline Dir: /gpfs/gsfs10/users/CCBR_Pipeliner/Pipelines/CARLISLE/.v2.4.1-a/bin
Git Commit/Tag: 57f7d79ff4bd5062ff776134d69346250328d148        v2.4.0-6-g57f7d79
Working Dir: /gpfs/gsfs12/users/sovacoolkl/carlisle_test

  #
  #
  #
    Folder /gpfs/gsfs12/users/sovacoolkl/carlisle_test already exists!
```

Perhaps instead it could throw a warning if the directory isn't empty?

### LINKED ISSUE #112: `carlisle --version` should print version number
`charlisle --version` does not print the version number

image

### LINKED ISSUE #113: rumode testrun produces error with DESEQ2
**Describe the bug**
Running the testrun runmode through the pipeline produces errors.

**To Reproduce**
```
bash carlisle --runmode=testrun --workdir=/data/sevillas2/carlisle
```

**Expected behavior**
The piepline should run to completion without this error.

### LINKED ISSUE #115: Improve runmode `testrun` by updating params, adding mm10
**Describe what enhancement you'd like to see.**
Need improvements to the testing environments
- [x] make the `tinytest` run the standard run (decrease contrast to one, samples to 5)
- [x] update defaults so only one q value is run, all peak callers, and both dedup and nondedup
~- [ ] create `testrun_hg38` and `testrun_mm10` runs~
~- [ ] update all test samples~

### LINKED ISSUE #116: add CHANGELOG
**Describe what enhancement you'd like to see.**
Add CHANGELOG.md to the pipeline

### LINKED ISSUE #118: RLibs Path for all RMD files
**Describe what enhancement you'd like to see.**
Currently the pipeline will install any libraries not found as needed. This is leading to permissions / installation issues. Instead the pipeline should use an RLib path set with all of the directories.

### LINKED ISSUE #121: Pipeline does not launch for mm10 due to missing parameters
The config file lists two parameters for hg38 that are not included for mm10:
    tss_bed:
    rose: 

These are required for rule rose, and the pipeline will not launch without them.

### LINKED ISSUE #125: GO enrichment is lagging with 'gopeaks' peak caller, during `linking distal enhancers`
**Describe the bug**
When running samples with `gopeaks` peak calling samples, GO Enrichment is lagging completion (> 16 hours) when running the `linking distal enhancers` step. This is compared with `seacr` and `macs2` samples which take approximately 5 hours total.
 
**Solution** 
Will skip this step when running `gopeaks` samples. Other informative information can be gleaned from other parts of the analysis.

### PR DESCRIPTION
**Pull request by slsevilla**
## Changes

- Adds a `--force` flag to allow for re-initialization of a workdir
- Fixes error with `testrun` in DESEQ2
- Decreases the number of samples being run with `testrun`, essentially running `tinytest` as default and removing `tinytest` as an option
- Reads version from `VERSION` file instead of github repo link
- Added a `CHANGELOG`
- Refactor R library handling
- Adds missing mm10 parameters
- Removes some testing from annotation report gopeaks due to latency

## Issues

- [x] fixes #96
- [x] fixes #97 
- [x] fixes #112
- [x] fixes #113 
- [x] fixes #115
- [x] fixes #116
- [x] fixes #118
- [x] fixes #121
- [x] fixes #125

## PR Checklist

(~Strikethrough~ any points that are not applicable.)

- [x] This comment contains a description of changes with justifications, with any relevant issues linked.
- [x] Update docs if there are any API changes.
- [x] Update `CHANGELOG.md` with a short description of any user-facing changes and reference the PR number. Guidelines: https://keepachangelog.com/en/1.1.0/

### CODE REVIEW

**Review by kopardev — APPROVED** [review]
Everything looks good to me. Also, delete the "test_dev" GHA WF as it refers to some 9999.9999 version which does not exist and include that in this PR.

**Comment by kelly-sovacool** [comment]
Can you edit the title of this PR to be a short description of the changes?

**Review by kelly-sovacool — CHANGES_REQUESTED** [review]
`testrun` should be switched to `runtest` in the docs, right?

https://github.com/CCBR/CARLISLE/blob/16a4d7a8f014941c2b73464bd08e0c34de129f58/docs/user-guide/run.md?plain=1#L14-L30

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+## CARLISLE development version
+
+- Fix: RNA report bug, caused by hard-coding of PC1-3, when only PC1-2 were generated (#104, @slsevilla)
+- Minor documentation improvements. (#100, @kelly-sovacool)
+- Fix: allow printing the version or help message even if singularity is not in the path. (#110, @kelly-sovacool)
+
```
[the remark this answers is not in the document]
Comment (slsevilla) [reply CHANGELOG.md:10]: yep - done with 42c8fd5f45bed26c026724cec5343fa8995238b2

**Comment by kelly-sovacool** [comment]
My test run failed in the GO enrichment rule:

```
Quitting from lines 19-52 [setup] (_go_enrichment.Rmd)
Error in `install.packages()`:
! unable to install packages
Backtrace:
  1. BiocManager::install(new.packages, INSTALL_opts = "--no-lock")
  2. BiocManager:::.install(...)
  3. BiocManager:::.install_repos(...)
  4. BiocManager:::.inet_install.packages(...)
  6. base::tryCatch(...)
  7. base (local) tryCatchList(expr, classes, parentenv, handlers)
  8. base (local) tryCatchOne(expr, names, parentenv, handlers[[1L]])
  9. value[[3L]](cond)
 10. BiocManager:::.inet_error(e)
Execution halted
```

The good news is all other jobs passed!

**Comment by slsevilla** [comment]
Finished testing these changes - everything went to completion. ready for you to test again @kelly-sovacool !

**Comment by kelly-sovacool** [comment]
test run submitted

```sh
bash /data/CCBR_Pipeliner/Pipelines/CARLISLE/carlisle-dev-sovacool/carlisle --runmode=init --workdir=/data/sovacoolkl/carlisle_test_iss-113_2
bash /data/CCBR_Pipeliner/Pipelines/CARLISLE/carlisle-dev-sovacool/carlisle --runmode=runtest --workdir=/data/sovacoolkl/carlisle_test_iss-113_2
```

**Review by kelly-sovacool — APPROVED** [review]
all jobs in my test run completed successfully 🎉

**Pull request merged**